tools/clang/tools/dxcompiler/DXCompiler.cpp

```diff
@@ -14,14 +14,17 @@
 #include "dxc/Support/Global.h"
 #include "dxc/Support/WinIncludes.h"
 #include "dxc/Support/HLSLOptions.h"
+#ifdef LLVM_ON_WIN32
 #include "dxcetw.h"
+#endif
 #include "dxillib.h"
 
 namespace hlsl { HRESULT SetupRegistryPassForHLSL(); }
 
 // C++ exception specification ignored except to indicate a function is not __declspec(nothrow)
 #pragma warning( disable : 4290 )
 
+#ifdef LLVM_ON_WIN32
```

Comment:
Sorry if this is obvious. But by wrapping these `new`/`delete` operators with `LLVM_ON_WIN32`, we are going through standard memory management on Unix systems and not going through DXC thread local mechanisms?

Reply:
That's right. Prior to this change, this file wasn't included anywhere but on Windows. I enabled it because Apple needs the initialization of the thread malloc and other static globals.  So this doesn't change any behavior in that regard. I've made some efforts to enable these, on Linux and did so in my original Linux port but haven't come up with something ironclad with the organization in the repo.

Reply:
+1